### PR TITLE
add option to install additional OS packages

### PR DESCRIPTION
**Pull request by SebastianAchilles**
This PR adds the option to install additional OS packages. By default no additional packages are installed. If needed, a list of OS packages to be installed can be added to `hieradata`.

### CODE REVIEW

**Comment by cmd-ntrf** [comment]
Would you see it as a problem if the packages were also installed on the management node?

If not, I would favor adding a new parameter to the class `profile::base` instead of defining a complete new class. We could move some the packages installed by `profile::base` in the default array, like vim for example.

**Comment by SebastianAchilles** [comment]
In a first attempt I added it to the class `profile::base`. Then I thought that it is not needed on the management node and changed it.

I am currently installing these additional OS packages in the cluster I am creating with Magic Castle for EasyBuild:
```
profile::additionalsoftware::packages:
  - python3-pip
  - Lmod
  - htop
  - tmux
  - bzip2
  - file
  - gcc-c++
  - make
  - openssl
  - openssl-devel
  - which
  - rdma-core-devel
  - diffutils
  - p7zip
  - p7zip-plugins
  - csh
  - glibc-static
  - perl-Thread-Queue
  - apptainer
```
These packages add up to an `Installed size: 362 M`. The extra space needed on the management node is not too much. So, I don't see it as a problem.

**Comment by cmd-ntrf** [comment]
Alright, then let's define a new parameter for profile::base instead of creating a new class. We will see with usage if the need to define packages to install with more granularity emerges.

To make sure the package added by the user are not already installed by another class, we should user `ensure_package` instead of a define a ` package` resource.

See https://www.puppetcookbook.com/posts/install-same-package-in-multiple-classes.html

**Comment by SebastianAchilles** [comment]
I have added the new parameter for `profile::base` and switched to `ensure_packages`. I tested this branch in our OpenStack Cloud and it worked as expected.

File: site/profile/manifests/base.pp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class profile::base (
   String $version,
   Optional[String] $admin_email = undef,
+  Array[String] $packages,
```
Comment (cmd-ntrf) [inline site/profile/manifests/base.pp:4]: Can you move this line before $admin_email?

**Comment by cmd-ntrf** [comment]
@SebastianAchilles : sorry for the delay, but I think we are almost ready to merge now. Can you fix the comment I have just left and then squash the commits into 1?

Thanks.

**Comment by mboisson** [comment]
I would be interested in this PR.

**Pull request merged**